code/modules/vehicles/speedbike.dm

```diff
@@ -36,6 +36,457 @@
+	F.Grant(M, src)
+	C.Grant(M, src)
+
+
+/obj/vehicle/space/speedbike/atmos/unbuckle_mob(mob/living/M)
+	. = ..()
+	if(ex_out)
+		M.transferItemToLoc(internal_extinguisher,src)
+	for(var/datum/action/innate/atmos_bike/H in M.actions)
+		qdel(H)
+
+/datum/action/innate/atmos_bike
+	check_flags = AB_CHECK_RESTRAINED | AB_CHECK_STUNNED | AB_CHECK_CONSCIOUS
+	var/obj/vehicle/space/speedbike/atmos/bike
+	var/obj/machinery/portable_atmospherics/scrubber/huge/inner_scrubber
+	var/obj/item/weapon/extinguisher/vehicle/VEX
+
+/datum/action/innate/atmos_bike/Grant(mob/living/L, obj/vehicle/B)
+	bike = B
+	inner_scrubber = bike.internal_scubber
+	VEX = bike.internal_extinguisher
+	..()
+
+/datum/action/innate/atmos_bike/Destroy()
+	bike = null
+	inner_scrubber = null
+	VEX = null
+	return ..()
+
+/datum/action/innate/atmos_bike/nanoice
+	name = "nanoice"
+	desc = "A potent anti-fire gas sprayed from your vehicle"
+	button_icon_state = "nanofrost"
+	var/nano_cooldown = 0
+
+/datum/action/innate/atmos_bike/nanoice/Activate()
+	if(world.time <= nano_cooldown)
+		owner << "<span class='warning'><b>Nanoice dispensers require another [round((nano_cooldown - world.time)/10)] seconds to recharge!</b></span>"
+		return
+	var/datum/effect_system/smoke_spread/freezing/S = new
+	S.set_up(2, owner.loc, blasting=1)
+	S.start()
+	var/obj/effect/decal/cleanable/flour/F = new /obj/effect/decal/cleanable/flour(owner.loc)
+	F.add_atom_colour("#B2FFFF", FIXED_COLOUR_PRIORITY)
+	F.name = "nanofrost residue"
+	F.desc = "Residue left behind from a nanofrost detonation. Perhaps there was a fire here?"
+	playsound(owner.loc,'sound/effects/bamf.ogg',100,1)
+	nano_cooldown = world.time + 300
+
+
+/datum/action/innate/atmos_bike/extinguish
+	name = "Arm the Extinguisher"
+	desc = "Unleashes a powerful fire extinguisher"
+	button_icon_state = "noflame"
+
+/datum/action/innate/atmos_bike/extinguish/Activate()
+	if(!bike.ex_out)
+		if(!owner.put_in_hands(VEX))
+			owner << "<span class='warning'>You need a free hand to hold the extinguisher!</span>"
+			return
+		VEX.loc = owner
+		bike.ex_out = TRUE
+		owner << "<span class='warning'>The vehicle unwinds a fire hose into your hands!</span>"
+		playsound(get_turf(owner),'sound/items/change_jaws.ogg', 75, 1)
+		name = "Store the Extinguisher"
+		desc = "Dropping the extinguisher will also automatically store it"
+		button_icon_state = "summons"
+		UpdateButtonIcon()
+		return
+	if(bike.ex_out)
+		owner.transferItemToLoc(VEX,bike)
+		name = "Arm the Extinguisher"
+		desc = "Unleashes a powerful fire extinguisher"
+		button_icon_state = "noflame"
+		bike.ex_out = FALSE
+		UpdateButtonIcon()
+
+
+/datum/action/innate/atmos_bike/scrub
+	name = "Scrubber Control"
+	desc = "Toggles the scrubber device built in to your vehicle"
+	button_icon_state = "mech_internals_on"
+
+/datum/action/innate/atmos_bike/scrub/Activate()
+	if(inner_scrubber.on)
+		owner << "<span class='notice'>You have disabled the vehicle's massive air scrubbers</span>"
```

Comment:
Most of SS13 notices are in second person present tense - this would normally be "You disable the vehicle's massive air scrubbers." To fit in even more, you might want to remove the "massive," since there aren't any other scrubbers on the vehicle.